src/components/pages/users-page/users-page.tsx

```diff
@@ -10,129 +12,326 @@ import {
+  function setAccessRequestWaiting(rowId: number | undefined, isLoading: boolean) {
+    let result: AccessRequestRow | undefined;
     setAccessRequests(requests => {
       return requests.map(request => {
-        if (request.id === activeAccessRequest?.id) {
-          request.waiting = false;
+        if (request.id === rowId) {
+          request.waiting = isLoading;
+          result = request;
         }
         return request;
       });
     });
+    return result;
+  }
+
+  function cancelRoleSelection() {
+    setAccessRequestWaiting(activeAccessRequest?.id, false);
     setActiveAccessRequest(undefined);
   }
 
-  async function acceptRoleSelection() {
-    setFinalizeApprovalLoading(true);
-    setFormDisabled(true);
-    if (activeAccessRequest) {
-      await axios.post(`api/access-request/${orgId}/approve`, {
-        requestId: activeAccessRequest.id,
-        roleId: availableRoles[selectedRole].id,
-      });
+  async function acceptRoleSelection(role: ApiRole) {
+    try {
+      setFinalizeApprovalLoading(true);
+      if (activeAccessRequest) {
+        await axios.post(`api/access-request/${orgId}/approve`, {
+          requestId: activeAccessRequest.id,
+          roleId: role.id,
+        });
+      }
+    } catch (error) {
+      showAlertDialog(error, 'Approve Access Request', 'Error while approving accessing request');
     }
     await initializeTable();
     setFinalizeApprovalLoading(false);
-    setFormDisabled(false);
     setActiveAccessRequest(undefined);
   }
 
   function acceptRequest(id: number) {
-    setAccessRequests(requests => {
-      return requests.map(request => {
-        if (request.id === id) {
-          request.waiting = true;
-        }
-        return request;
-      });
-    });
-    const request = accessRequests.find(req => req.id === id);
-    if (request) {
-      setActiveAccessRequest(request);
-    }
+    setActiveAccessRequest(setAccessRequestWaiting(id, true));
   }
 
   async function denyRequest(id: number) {
-    updateDenyRequestLoading(id, true);
-    setAccessRequests(requests => {
-      return requests.map(request => {
-        if (request.id === id) {
-          request.waiting = true;
-        }
-        return request;
+    try {
+      updateDenyRequestLoading(id, true);
+      await axios.post(`api/access-request/${orgId}/deny`, {
+        requestId: id,
       });
-    });
-    await axios.post(`api/access-request/${orgId}/deny`, {
-      requestId: id,
-    });
-    updateDenyRequestLoading(id, false);
+    } catch (error) {
+      showAlertDialog(error, 'Deny Access Request', 'Error while denying accessing request');
+    }
     await initializeTable();
+    updateDenyRequestLoading(id, false);
   }
 
-  const handleAlertClose = () => {
-    setAlert({ open: false, message: '', title: '' });
-  };
-
-  useEffect(() => { initializeTable().then(); }, [initializeTable]);
+  useEffect(() => { initializeTable(); }, [initializeTable]);
```

Comment:
We have been using `.then()` for these to avoid IDE warnings about unhandled promises.

Reply:
Strange. I don't think I've seen those warnings. Maybe we should sync on how to get my environment set up the same as yours? I'll add this back in.